streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -1522,6 +1525,72 @@ private void maybeScheduleFollowupRebalance(final long encodedNextScheduledRebal
         }
     }
 
+    private AssignmentError validateTaskAssignment(final ApplicationState applicationState,
+                                                   final TaskAssignment taskAssignment) {
+        final Collection<KafkaStreamsAssignment> assignments = taskAssignment.assignment();
+        final Set<TaskId> activeTasksInOutput = new HashSet<>();
+        final Set<TaskId> standbyTasksInOutput = new HashSet<>();
+        for (final KafkaStreamsAssignment assignment : assignments) {
+            final Set<TaskId> tasksForAssignment = new HashSet<>();
+            for (final KafkaStreamsAssignment.AssignedTask task : assignment.assignment()) {
+                if (activeTasksInOutput.contains(task.id()) && task.type() == KafkaStreamsAssignment.AssignedTask.Type.ACTIVE) {
+                    log.error("Assignment is invalid: an active task was assigned multiple times: {}", task.id());
```

Comment:
It would be nice to include the ProcessId of both clients. We'll need to change `activeTasksInOutput` into a Map<TaskId, ProcessId> so we can look up the other ProcessId, then do something like this:
```suggestion
                    log.error("Assignment is invalid: active task {} was assigned to multiple KafkaStreams clients: {} and {}", task.id(), assignments.processId().id(), activeTasksInOutput.get(task.id().id()));
```